docs/Guidelines.md

```diff
@@ -0,0 +1,78 @@
+# Overview
+The purpose of *this* document is to serve as a primer for developers/qe who are looking for best practices or adding new tests to RHTAP E2E framework.
+
+## General tips
+
+* Make sure you've implemented any required controller functionality that is required for your tests within the following files
+   * `pkg/utils/<new controller directory>` - net new controller logic for a new service or component
+   * `pkg/framework/framework.go` - import the new controller and update the `Framework` struct to be able to initialize the new controller
+* Every test package should be imported to `cmd/e2e_test.go`, e.g. [has](https://github.com/redhat-appstudio/e2e-tests/blob/main/cmd/e2e_test.go#L15).
+* Every new test should have correct [labels](docs/LabelsNaming.md).
+* Every test should have meaningful description with JIRA/GitHub issue key.
+* (Recommended) Use JIRA integration for linking issues and commits (just add JIRA issue key in the commit message).
+* When running via mage you can filter the suites run by specifying the
+  `E2E_TEST_SUITE_LABEL` environment variable. For example:
+  `E2E_TEST_SUITE_LABEL=ec ./mage runE2ETests`
+* `klog` level can be controled via `KLOG_VERBOSITY` environment variable. For
+  example: `KLOG_VERBOSITY=9 ./mage runE2ETests` would output `curl` commands
+  issued via Kubernetes client from sigs.k8s.io/controller-runtime
+* To quickly debug a test, you can run only the desired suite. Example: `./bin/e2e-appstudio --ginkgo.focus="e2e-demos-suite"`
+* Split tests in different scenarios. Will be better to debug a small scenario than a very big one
+
+## Debuggability
+
+If your test fails, it should provide as detailed as possible reasons for the failure in its failure message. The failure message is the string that gets passed (directly or indirectly) to ginkgo.Fail[f].
+
+A good failure message:
+* identifies the test failure
+* has enough details to provide some initial understanding of what went wrong
+
+It is good practice to include details like the object that failed some assertion in the failure message because then a) the information is available when analyzing a failure that occurred in the CI and b) it only gets logged when some assertion fails. Always dumping objects via log messages can make the test output very large and may distract from the relevant information.
+
+Dumping structs with `format.Object` is recommended. Starting with Kubernetes 1.26, format.Object will pretty-print Kubernetes API objects or structs as YAML and omit unset fields, which is more readable than other alternatives like `fmt.Sprintf("%+v")`.
+
+```golang
+import (
+    "fmt"
+    "k8s.io/api/core/v1"
+    "k8s.io/kubernetes/test/utils/format"
+)
+
+var pod v1.Pod
+fmt.Printf("format.Object:\n%s", format.Object(pod, 1 /* indent one level */))
+
+format.Object:
```

Comment:
Will it be good to keep the output of the code outside the code block?

Reply:
in this case is irrelevant, is an example of how code works